router/protect.go

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2021 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package router
+
+import (
+	"net"
+	"sync"
+)
+
+var (
+	mu      sync.RWMutex
+	protect func(fd int) error = nil
+)
+
+// SetProtectFunc sets the callback for using to protect provided connection from going through the tunnel.
+func SetProtectFunc(f func(fd int) error) {
+	mu.Lock()
```

Comment:
Regarding this synchronization pattern, I would to ask: are we resetting this `protect` function often or is just a one-off?

Reply:
This happens only once per session.